Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -339,5 +424,108 @@ protected override void OnGotFocus(RoutedEventArgs e)
             _parentMenu.SelectedMenuItem = this;
             base.OnGotFocus(e);
         }
+
+        internal void Underline()
+        {
+            if (_originalHeader == null)
+            {
+                return;
+            }
+
+            var underlineCharacterIndex = _originalHeader.IndexOf(UnderlineCharacter);
+
+            var underlinedCharacter = _originalHeader[underlineCharacterIndex + 1];
+            var text = new TextBlock();
+
+            var runWithUnderlinedCharacter = new Run
+            {
+                Text = underlinedCharacter.ToString()
+            };
+
+            if (_isTextTextDecorationsSupported)
+            {
+                runWithUnderlinedCharacter.TextDecorations = Windows.UI.Text.TextDecorations.Underline;
+            }
+
+            var firstPartBuilder = new StringBuilder();
+            var secondPartBuilder = new StringBuilder();
+
+            for (int i = 0; i < underlineCharacterIndex; i++)
+            {
+                firstPartBuilder.Append(_originalHeader[i]);
+            }
+
+            for (int i = underlineCharacterIndex + 2; i < _originalHeader.Length; i++)
+            {
+                secondPartBuilder.Append(_originalHeader[i]);
+            }
+
+            var firstPart = firstPartBuilder.ToString();
+            var secondPart = secondPartBuilder.ToString();
+
+            if (!string.IsNullOrEmpty(firstPart))
+            {
+                text.Inlines.Add(new Run
+                {
+                    Text = firstPart
+                });
+            }
+
+            text.Inlines.Add(runWithUnderlinedCharacter);
+
+            if (!string.IsNullOrEmpty(secondPart))
+            {
+                text.Inlines.Add(new Run
+                {
+                    Text = secondPart
+                });
+            }
+
+            InternalHeader = text;
+        }
+
+        private static void HeaderPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+        {
+            var menuItem = (MenuItem)sender;
+
+            if (menuItem._isInternalHeaderUpdate)
+            {
+                menuItem._isInternalHeaderUpdate = false;
+                return;
+            }
+
+            menuItem._originalHeader = null;
+
+            var headerString = menuItem.Header as string;
```

Comment:
we can included this in a next release as I'll not be available for creating new features until weekend (after code freeze).